base/client.jl

```diff
@@ -210,120 +208,126 @@ function init_bind_addr(args::Vector{UTF8String})
+            println(STDERR, "julia: option `$arg` is missing an argument")
+            exit(1)
+        end
+        repl = true
+        startup = true
+        quiet = false
+        color_set = false
+        no_history_file = false
+        while true
+            # show julia VERSION and quit
+            if bool(opts.version)
+                println(STDOUT, "julia version ", VERSION)
+                exit(0)
             end
-            addprocs(np)
-        elseif args[i]=="--machinefile"
-            i == length(args) && error("--machinefile  no <file> provided")
-            i+=1
-            machines = load_machine_file(args[i])
-            addprocs(machines)
-        elseif args[i]=="-v" || args[i]=="--version"
-            println("julia version ", VERSION)
-            exit(0)
-        elseif args[i]=="--no-history"
-            # deprecated in v0.3
-            warn("'--no-history' is deprecated; use '--no-history-file'")
-            no_history_file = true
-        elseif args[i] == "--no-history-file"
-            no_history_file = true
-        elseif args[i] == "-f" || args[i] == "--no-startup"
-            startup = false
-        elseif args[i] == "-F"
-            # load juliarc now before processing any more options
-            load_juliarc()
-            startup = false
-        elseif args[i] == "-i"
-            global is_interactive = true
-        elseif startswith(args[i], "--color")
-            if args[i] == "--color"
-                color_set = true
-                global have_color = true
-            elseif args[i][8] == '='
-                val = args[i][9:end]
-                if in(val, ("no","0","false"))
-                    color_set = true
-                    global have_color = false
-                elseif in(val, ("yes","1","true"))
-                    color_set = true
-                    global have_color = true
+            # startup worker
+            if bool(opts.worker)
+                assert(opts.worker == 1 || opts.worker == 2)
+                # if default, start worker process otherwise if custom pass through
+                if opts.worker == 1
+                    start_worker() # does not return
                 end
             end
-            if !color_set
-                error("invalid option: ", args[i])
+            # load file immediately on all processors
+            if opts.load != C_NULL
+                require(bytestring(opts.load))
             end
-        elseif args[i][1]!='-'
-            if startup
+            # show banner
+            quiet = bool(opts.quiet)
+            # load ~/.juliarc file
+            if opts.startupfile == 1
                 load_juliarc()
                 startup = false
+            elseif opts.startupfile == 2
+                startup = false
+            end
+            # load ~/.julia_history file
+            no_history_file = bool(opts.historyfile)
```

Comment:
I switched this to `history_file=true/false` which follows other changes in this PR.